server.go

```diff
@@ -495,8 +507,10 @@ func (s *server) inboundPeerConnected(conn net.Conn) {
 	// cancel the ongoing connection attempts to ensure that we don't end
 	// up with a duplicate connecting to the same peer.
 	s.pendingConnMtx.RLock()
-	if connReq, ok := s.persistentConnReqs[pubStr]; ok {
-		s.connMgr.Remove(connReq.ID())
+	if connReqs, ok := s.persistentConnReqs[pubStr]; ok {
+		for _, connReq := range connReqs {
```

Comment:
Not totally sure how this works, but when I added this to outboundPeerConnected, the nodes didn't reconnect upon startup. Is there something else I have to do to make that happen?